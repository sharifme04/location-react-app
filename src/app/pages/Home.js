import React, { useEffect, useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import {
  Header,
  HeaderLayout,
  Breadcrumb,
  SearchLocationCard,
  SuggestionCard,
} from "../components";
import { ReactComponent as HeaderSliderImageSvg } from "../assets/images/header-slider-image-1.svg";
import { locationsPednding } from "../actions/locations";
import { tripSearchPednding } from "../actions/tripSearch";

const Home = () => {
  const dispatch = useDispatch();
  const [locationPoint, setLocationPoint] = useState("");
  const [display, setDisplay] = useState("");
  const [values, setValues] = useState({
    startPoint: "",
    endPoint: "",
  });

  const handleSuggestion = (value) => setDisplay(value);

  const [idValues, setIdValues] = useState({
    startPointId: "",
    endPointId: "",
  });

  const { locations } = useSelector((state) => state.locationsReducer || []);

  useEffect(() => {
    values[locationPoint] && dispatch(locationsPednding(values[locationPoint]));
  }, [values[locationPoint], dispatch]); // eslint-disable-line react-hooks/exhaustive-deps

  const handleChange = (event) => {
    const { name, value } = event.target;
    setValues({ ...values, [name]: value });
  };

  const handlebothLocation = (e) => {
    setValues({ ...values, [locationPoint]: e.objectName + ", " + e.locality });
    setIdValues({ ...idValues, [[locationPoint] + "Id"]: e.id });
  };

  const handleLocationPoint = (value) => {
    setLocationPoint(value);
  };

  const handetripSearch = () => {
    idValues?.startPointId &&
      idValues?.endPointId &&
      dispatch(tripSearchPednding(idValues.startPointId, idValues.endPointId));
  };

  return (
    <div>
      <HeaderLayout>
        <Breadcrumb />
        <Header
          title={"Umweltrechner"}
          subt1={"Umwelt und Preisvergleich zwischen VVS und Deinem Auto"}
        />
      </HeaderLayout>
      <div className="slide-and-search">
        <div className="slide">
          <HeaderSliderImageSvg />
        </div>
        <SearchLocationCard
          values={values}
          handleSuggestion={handleSuggestion}
          handleChange={handleChange}
          handleLocationPoint={handleLocationPoint}
          handetripSearch={handetripSearch}
        />
      </div>
      <SuggestionCard
        locations={locations}
        display={display}
        handleSuggestion={handleSuggestion}
        handlebothLocation={handlebothLocation}
      />
    </div>
  );
};

export default Home;
