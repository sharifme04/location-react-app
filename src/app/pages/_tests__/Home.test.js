import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import { MemoryRouter } from "react-router-dom";
import Home from "../Home";

const mockStore = configureMockStore();
const store = mockStore({});

test("snapshot form renders", () => {
  const component = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    </Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});