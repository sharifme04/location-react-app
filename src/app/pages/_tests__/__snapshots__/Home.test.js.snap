// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot form renders 1`] = `
<div>
  <div
    className="headerContainer"
  >
    <div>
      <nav
        aria-label="breadcrumb"
      >
        <ol
          className="breadcrumb"
        >
          <li
            className="breadcrumb-item"
          >
            Rundum mobil
          </li>
          <li
            className="breadcrumb-item"
          >
            Preis- und Umweltvergleich
          </li>
        </ol>
      </nav>
    </div>
    <div>
      <div>
        <span
          className="title"
        >
          Umweltrechner
        </span>
      </div>
      <div
        className="subtitle"
      >
        <span>
          Umwelt und Preisvergleich zwischen VVS und Deinem Auto
        </span>
      </div>
    </div>
  </div>
  <div
    className="slide-and-search"
  >
    <div
      className="slide"
    >
      <svg>
        header-slider-image-1.svg
      </svg>
    </div>
    <div
      className="search-location"
    >
      <div
        className="card-title"
      >
        WÃ¤hle Deine Pendelroute 
      </div>
      <div>
        <p
          className="card-subtitle"
        >
          um zu sehen, wie viel CO2 Du mit VVS sparen kannst
        </p>
      </div>
      <div
        className="input-container"
      >
        <div
          className="icons-column"
        >
          <svg
            className="top-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 56c110.532 0 200 89.451 200 200 0 110.532-89.451 200-200 200-110.532 0-200-89.451-200-200 0-110.532 89.451-200 200-200m0-48C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm0 168c-44.183 0-80 35.817-80 80s35.817 80 80 80 80-35.817 80-80-35.817-80-80-80z"
            />
          </svg>
          <div
            className="vertical-line"
          />
          <svg
            className="bottom-icon"
            fill="currentColor"
            height={20}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width={20}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 32C167.67 32 96 96.51 96 176c0 128 160 304 160 304s160-176 160-304c0-79.49-71.67-144-160-144zm0 224a64 64 0 1164-64 64.07 64.07 0 01-64 64z"
            />
          </svg>
        </div>
        <div
          className="input-group"
        >
          <input
            className="input-style"
            name="startPoint"
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="von : Ort, Haltestelle, Adresse, POI"
            value=""
          />
          <input
            className="input-style"
            name="endPoint"
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="von : Ort, Haltestelle, Adresse, POI"
            value=""
          />
        </div>
        <div
          className="right-arrow"
        >
          <div />
          <div>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M11.95 7.95l-1.414 1.414L8 6.828 8 20H6V6.828L3.465 9.364 2.05 7.95 7 3l4.95 4.95zm10 8.1L17 21l-4.95-4.95 1.414-1.414 2.537 2.536L16 4h2v13.172l2.536-2.536 1.414 1.414z"
                />
              </g>
            </svg>
          </div>
          <div />
        </div>
      </div>
      <button
        className="weiterBtn"
        onClick={[Function]}
      >
        Weiter
      </button>
    </div>
  </div>
</div>
`;
